core/domain/activity_domain.py

```diff
@@ -28,17 +28,38 @@ class ActivityReference(object):
     """
 
     def __init__(self, activity_type, activity_id):
+        """Constructs an ActivityReference instance.
+
+        Args:
+          activity_type: str.
+          activity_id:   str.
+        """
         self.type = activity_type
         self.id = activity_id
 
     def get_hash(self):
+        """Gets instance hash.
```

Comment:
Actually, I took a look, and it turns out it's used in one place. The key insight here is that the hash is a string that uniquely identifies the activity reference (so it can be used for checking duplicates etc.). So this docstring might become:

    Returns a unique string that uniquely identifies this ActivityReference domain object.